src/server/job_func.c

```diff
@@ -906,14 +904,12 @@ job_purge(job *pjob)
 	}
 #else
 	/* delete job and dependants from database */
-	if (!(pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob)) {
-		obj.pbs_db_obj_type = PBS_DB_JOB;
-		obj.pbs_db_un.pbs_db_job = &dbjob;
-		strcpy(dbjob.ji_jobid, pjob->ji_qs.ji_jobid);
-		if (pbs_db_delete_obj(conn, &obj) == -1) {
-			log_joberr(-1, __func__, msg_err_purgejob_db,
-				pjob->ji_qs.ji_jobid);
-		}
+	obj.pbs_db_obj_type = PBS_DB_JOB;
```

Comment:
Does this mean job dependencies are now supported for subjobs or ranges of subjobs as well ?

Reply:
Not sure we want to support job dependencies for sub-jobs. If it works fine but we should not document it as a supported feature.

Reply:
Actually contents of this function of file does not handle job dependency operations.  All the job dependency operations are done in file: https://github.com/PBSPro/pbspro/blob/master/src/server/req_register.c . 
This PR does not have any changes in that file and I have not altered any code explicitly so as to enable dependency on subjob or range of subjobs.

However I will add to my pre merge checklist to check if there is any impact on job dependency feature due to this PR

Reply:
Looks like the author of the comment wanted to mean job db related dependents like job attributes and job script. and did not actually mean dependent job which is handled in reg_register.c